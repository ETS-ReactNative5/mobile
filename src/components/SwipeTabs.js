import React, { Component } from 'react'
import {
  StyleSheet,
  TouchableOpacity,
  View
} from 'react-native'
import EStyleSheet from 'react-native-extended-stylesheet'
import StyledText from './StyledText'
import FieldGuide from './FieldGuide'
import Icon from 'react-native-vector-icons/FontAwesome'
import { length } from 'ramda'

export class SwipeTabs extends Component {
  constructor(props) {
    super(props)

    this.state = {
      isFieldGuideVisible: false,
    }
  }
  render() {
    const fieldGuideButton =
      <TouchableOpacity
        onPress={() => this.setState({isFieldGuideVisible: true})}
        activeOpacity={0.5}
        style={ styles.button }>
        <Icon name='map-o' style={styles.icon} />
        <StyledText additionalStyles={[styles.buttonText]} text={ 'Field Guide' } />
      </TouchableOpacity>

    const fieldGuide =
      <FieldGuide
        guide={this.props.guide}
        isVisible={this.state.isFieldGuideVisible}
        onClose={() => this.setState({isFieldGuideVisible: false})}
      />

    return (
      <View>
        <View style={styles.container}>
          { length(this.props.guide.items) > 0 ? fieldGuideButton : null }
        </View>
        { this.state.isFieldGuideVisible ? fieldGuide : null }
      </View>
    )
  }
}

const styles = EStyleSheet.create({
  $containerHeight: 60,
  container: {
    backgroundColor: '$lightestGrey',
    position: 'absolute',
    bottom: 0,
    left: 0,
    right: 0,
    height: '$containerHeight',
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'space-around'
  },
  button: {
    backgroundColor: 'white',
    borderColor: '$disabledIconColor',
    borderWidth: StyleSheet.hairlineWidth,
    borderRadius: 5,
    paddingVertical: 8,
    paddingHorizontal: 12,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center',
    margin: 10,
  },
  buttonText: {
    color: '$darkGrey',
  },
  icon: {
    fontSize: 16,
    color: '$darkGrey',
    padding: 3,
    marginRight: 5
  },
})

SwipeTabs.propTypes = {
  guide: React.PropTypes.object,
}

export default SwipeTabs
