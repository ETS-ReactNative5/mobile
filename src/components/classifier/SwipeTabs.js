import React, { Component } from 'react'
import {
  StyleSheet,
  TouchableOpacity,
  View
} from 'react-native'
import PropTypes from 'prop-types';
import EStyleSheet from 'react-native-extended-stylesheet'
import FontedText from '../common/FontedText'
import FieldGuide from './FieldGuide'
import { length } from 'ramda'

export class SwipeTabs extends Component {
  constructor(props) {
    super(props)

    this.state = {
      isFieldGuideVisible: false,
    }
  }
  render() {
    const leftButton =
      <TouchableOpacity
        onPress={this.props.onLeftButtonPressed}
        activeOpacity={0.5}
        style={ [styles.button, styles.tealButton, styles.leftButtonPadding] }>
        <FontedText style={[styles.buttonText, styles.tealButtonText]}>
          { this.props.answers[0].label }
        </FontedText>
      </TouchableOpacity>


    const rightButton =
      <TouchableOpacity
        onPress={this.props.onRightButtonPressed}
        activeOpacity={0.5}
        style={ [styles.button, styles.tealButton] }>
        <FontedText style={[styles.buttonText, styles.tealButtonText]}>
          { this.props.answers[1].label }
        </FontedText>
      </TouchableOpacity>

    const fieldGuideButton =
      <TouchableOpacity
        onPress={() => this.setState({isFieldGuideVisible: true})}
        activeOpacity={0.5}
        style={ [styles.buttonShadow, styles.button, styles.leftButtonPadding] }>
        <FontedText style={styles.buttonText}>
           Field Guide
        </FontedText>
      </TouchableOpacity>

    const fieldGuide =
      <FieldGuide
        guide={this.props.guide}
        isVisible={this.state.isFieldGuideVisible}
        onClose={() => this.setState({isFieldGuideVisible: false})}
      />

    return (
      <View>
        <View style={styles.container}>
          { leftButton }
          { length(this.props.guide.items) > 0 ? fieldGuideButton : null }
          { rightButton }
        </View>
        { this.state.isFieldGuideVisible ? fieldGuide : null }
      </View>
    )
  }
}

const styles = EStyleSheet.create({
  container: {
    marginHorizontal: 25,
    marginVertical: 15,
    backgroundColor: 'transparent',
    flexDirection: 'row',
    justifyContent: 'space-around'
  },
  leftButtonPadding: {
    marginRight: 20
  },
  button: {
    backgroundColor: 'white',
    borderColor: '$disabledIconColor',
    borderWidth: StyleSheet.hairlineWidth,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center',
  },
  buttonShadow: {
    shadowColor: 'rgba(0, 0, 0, 0.5)',
    shadowOpacity: 1,
    shadowRadius: 2,
    shadowOffset: {
        height: 1,
        width: 2,
    },
  },
  buttonText: {
    marginVertical: 11,
    marginHorizontal: 9
  },
  tealButton: {
    flex: 1,
    backgroundColor: '$buttonColor'
  },
  tealButtonText: {
    color: 'white', 
  },
})

SwipeTabs.propTypes = {
  guide: PropTypes.object,
  onLeftButtonPressed: PropTypes.func,
  onRightButtonPressed: PropTypes.func,
  answers: PropTypes.arrayOf(PropTypes.object)
}

export default SwipeTabs
