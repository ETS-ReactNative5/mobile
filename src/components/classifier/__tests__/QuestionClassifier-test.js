import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'
import SwipeTabs from '../SwipeTabs'

const guide = {
    items: [
        {
            title: 'fake_avatar',
            content: 'Nice project',
            icon: '12345',
        },
        {
            title: 'fake_avatar2',
            content: 'Another Nice project',
            icon: '123456',
        },
    ],
    icons:{
        '12345': { src: 'fake_icon.jpg' },
        '123456': { src: 'fake_icon.jpg' }
    }
}

const answers=[]

it('renders correctly and with no button if no field guide', () => {
    const tree = renderer.create(
        ={jest.fn} answers={answers} />
    ).toJSON()
    expect(tree).toMatchSnapshot()
})

