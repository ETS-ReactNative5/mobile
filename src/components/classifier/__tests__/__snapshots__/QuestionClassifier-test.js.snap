// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Context.Provider
  value={
    Object {
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <UNDEFINED
    classifierActions={
      Object {
        "setClassifierTestMode": [Function],
        "startNewClassification": [Function],
      }
    }
    isFetching={false}
    navBarActions={
      Object {
        "setColorForPageToDefault": [Function],
        "setTitleForPage": [Function],
      }
    }
    project={
      Object {
        "display_name": "Awesome project",
      }
    }
    saveAnnotation={[Function]}
    saveClassification={[Function]}
    seenThisSession={Array []}
    setIsFetching={[Function]}
    startNewClassification={[Function]}
    subject={
      Object {
        "display": Object {
          "src": "blah.jpg",
        },
        "id": "23432432",
      }
    }
    subjectSizes={
      Object {
        "resizedHeight": 100,
        "resizedWidth": 100,
      }
    }
    task={
      Object {
        "answers": Array [
          Object {
            "label": "Yes",
          },
          Object {
            "label": "No",
          },
          Object {
            "label": "Depends if I am hungry",
          },
        ],
        "question": "Do you like this question?",
      }
    }
    workflow={
      Object {
        "configuration": Object {
          "pan_and_zoom": true,
        },
        "first_task": "T0",
        "tasks": Object {
          "T0": Object {
            "answers": Array [
              Object {
                "label": "Yes",
              },
              Object {
                "label": "No",
              },
              Object {
                "label": "Depends if I am hungry",
              },
            ],
            "question": "Do you like this question?",
          },
        },
      }
    }
    workflowID="1"
  />
</Context.Provider>
`;

exports[`renders spinner if fetching 1`] = `
<Context.Provider
  value={
    Object {
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <UNDEFINED
    classifierActions={
      Object {
        "setClassifierTestMode": [Function],
        "startNewClassification": [Function],
      }
    }
    isFetching={true}
    navBarActions={
      Object {
        "setColorForPageToDefault": [Function],
        "setTitleForPage": [Function],
      }
    }
    project={
      Object {
        "display_name": "Awesome project",
      }
    }
    seenThisSession={Array []}
    setIsFetching={[Function]}
    startNewClassification={[Function]}
    subject={
      Object {
        "display": Object {
          "src": "blah.jpg",
        },
        "id": "23432432",
      }
    }
    subjectSizes={
      Object {
        "resizedHeight": 100,
        "resizedWidth": 100,
      }
    }
    task={
      Object {
        "answers": Array [
          Object {
            "label": "Yes",
          },
          Object {
            "label": "No",
          },
          Object {
            "label": "Depends if I am hungry",
          },
        ],
        "question": "Do you like this question?",
      }
    }
    workflow={
      Object {
        "configuration": Object {
          "pan_and_zoom": true,
        },
        "first_task": "T0",
        "tasks": Object {
          "T0": Object {
            "answers": Array [
              Object {
                "label": "Yes",
              },
              Object {
                "label": "No",
              },
              Object {
                "label": "Depends if I am hungry",
              },
            ],
            "question": "Do you like this question?",
          },
        },
      }
    }
    workflowID="1"
  />
</Context.Provider>
`;

exports[`renders tutorial if needed 1`] = `
<Context.Provider
  value={
    Object {
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <UNDEFINED
    classifierActions={
      Object {
        "setClassifierTestMode": [Function],
        "startNewClassification": [Function],
      }
    }
    isFetching={false}
    navBarActions={
      Object {
        "setColorForPageToDefault": [Function],
        "setTitleForPage": [Function],
      }
    }
    needsTutorial={true}
    project={
      Object {
        "display_name": "Awesome project",
      }
    }
    seenThisSession={Array []}
    setIsFetching={[Function]}
    startNewClassification={[Function]}
    subject={
      Object {
        "display": Object {
          "src": "blah.jpg",
        },
        "id": "23432432",
      }
    }
    subjectSizes={
      Object {
        "resizedHeight": 100,
        "resizedWidth": 100,
      }
    }
    task={
      Object {
        "answers": Array [
          Object {
            "label": "Yes",
          },
          Object {
            "label": "No",
          },
          Object {
            "label": "Depends if I am hungry",
          },
        ],
        "question": "Do you like this question?",
      }
    }
    workflow={
      Object {
        "configuration": Object {
          "pan_and_zoom": true,
        },
        "first_task": "T0",
        "tasks": Object {
          "T0": Object {
            "answers": Array [
              Object {
                "label": "Yes",
              },
              Object {
                "label": "No",
              },
              Object {
                "label": "Depends if I am hungry",
              },
            ],
            "question": "Do you like this question?",
          },
        },
      }
    }
    workflowID="1"
  />
</Context.Provider>
`;
